### ISSUE TITLE
Support for Lenovo IdeaPad Gaming 3 

### ISSUE THREAD

**Issue by CygnusTerminal** [0xless/battmngr#5]
Laptop model: Gaming 3 15ARH05
Output of dmidecode: 82EY
ideapad-cm works for enabling/disabling conservation mode.
ACPI dump: https://controlc.com/5e2e61fd

**Comment by CygnusTerminal** [comment]
Sorry, tried and seems to work fine!

**Comment by CygnusTerminal** [comment]
EDIT: Setting battery mode does not work.
`Can't update the battery mode, please retry`

**Comment by 0xless** [comment]
Thanks for opening this issue, `battmngr` works using ACPI calls only, while `ideapad-cm` uses ideapad_acpi drivers (/sys/bus/platform/drivers/ideapad_acpi/) and can offer less features.

If you want to contribute further, it would be amazing to have a detailed list of commands/features working, not working or showing weird behaviors. In this way I can mark your laptop as partially supported and work on a full support!

**Comment by CygnusTerminal** [comment]
> Thanks for opening this issue, `battmngr` works using ACPI calls only, while `ideapad-cm` uses ideapad_acpi drivers (/sys/bus/platform/drivers/ideapad_acpi/) and can offer less features.
> 
> If you want to contribute further, it would be amazing to have a detailed list of commands/features working, not working or showing weird behaviors. In this way I can mark your laptop as partially supported and work on a full support!

Thanks! So, i just install `battmngr` and try every command, right? Then i just post what works and what does not work. Is that correct?

**Comment by 0xless** [comment]
That's correct! Please include any inconsistent behavior or unexpected errors, thanks!

**Comment by CygnusTerminal** [comment]
Sorry for the late answer:
battmngr -s does not work for setting cooling modes (error: Can't update the battery mode, please retry) dmesg logs show this:
[ 2987.328368] acpi_call: Cannot get handle: Error: AE_NOT_FOUND
journalctl logs:
Oct 12 15:54:19 blackfuture kernel: acpi_call: Cannot get handle: Error: AE_NOT_FOUND
Reading cooling modes work. Setting either rapid charge/conservation mode works.
EDIT: To clarify, the time appearing in the journalctl log corresponds to the time the command was issued.

**Comment by 0xless** [comment]
I'm sorry for the late answer myself, so, that error seem to pop up when an invalid/non supported ACPI call is made.
That means that your laptop is not compatible with Ideapad 5are05 calls.

At the moment I haven't researched much, but it should be possible to monitor ACPI calls made with Lenovo Vantage under Windows and replay those using battmngr in a Linux environment. 

If you could provide us with a list of each supported command and their relative output, I can compile a table and update the README file so that new users will have more info.

Once again, thanks for the interest in this project and thanks for spending time supporting it!

**Comment by CygnusTerminal** [comment]
> I'm sorry for the late answer myself, so, that error seem to pop up when an invalid/non supported ACPI call is made. That means that your laptop is not compatible with Ideapad 5are05 calls.
> 
> At the moment I haven't researched much, but it should be possible to monitor ACPI calls made with Lenovo Vantage under Windows and replay those using battmngr in a Linux environment.
> 
> If you could provide us with a list of each supported command and their relative output, I can compile a table and update the README file so that new users will have more info.
> 
> Once again, thanks for the interest in this project and thanks for spending time supporting it!
Right, i'll dualboot Windows later this week and check if i can get the acpi calls from Vantage and i'll upload them here!
